src/course-page-resources/course-page/CoursePageConfigCard.jsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+import PropTypes from 'prop-types';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faCog } from '@fortawesome/free-solid-svg-icons';
+import { Button } from '@edx/paragon';
+import classNames from 'classnames';
+
+import messages from '../messages';
+
+function CoursePageConfigCard({
+  intl,
+  id,
+  title,
+  description,
+  pageStatus,
+  showSettings,
+}) {
+  const cardSettings = 'bg-white shadow p-3 m-0 my-3 course-page-title';
+  const cardClasses = classNames('card', cardSettings, { disabled: !pageStatus });
+  const cardSubSettings = 'bg-white px-3 py-0';
+
+  const buttonSettings = 'd-flex mt-0 justify-content-center align-items-center';
+  const buttonClasses = classNames(buttonSettings, { visible: !pageStatus, invisible: pageStatus });
+  const pageStatusMsgId = pageStatus ? 'pageStatus.enabled' : 'pageStatus.disabled';
+
+  return (
+    <div className={cardClasses} key={id}>
+      <div className={classNames('card-header', cardSubSettings)}>
+        <div className="row">
+          <h2 className="font-weight-bold">{title}</h2>
+          <div className="col flex-grow" />
+          {showSettings && <FontAwesomeIcon icon={faCog} />}
+        </div>
+      </div>
+      <div className={classNames('card-body', cardSubSettings)}>
+        {showSettings && <p>{intl.formatMessage(messages[pageStatusMsgId])}</p>}
+        <p>{description}</p>
+      </div>
+      <div className={classNames('card-footer', cardSubSettings)}>
+        <div className={buttonClasses}>
+          {!showSettings && (
+            <Button className="course-page-button bg-white border font-weight-bold rounded">
+              {intl.formatMessage(messages['enable.button'])}
+            </Button>
+          )}
+        </div>
+      </div>
+    </div>
+  );
+}
+
+CoursePageConfigCard.propTypes = {
+  intl: intlShape.isRequired,
+  id: PropTypes.string,
+  title: PropTypes.string,
+  description: PropTypes.string,
+  pageStatus: PropTypes.bool,
```

Comment:
This is very ambiguous. What status is this? It seems to signify that the page/app is enabled.

Reply:
From examining the various combinations shown in the wire-frame, that is my understanding of what is going on: the page can be added (possibly in a disabled state) and then later enabled and disabled at will. This is why I have lumped the behavior of several seemingly unrelated items into "showSettings".

I thought I had asked for specs on actual behavior (not just appearance), but I can't find the request. Knowing how things are meant to behave would help immensely by doing away with a lot of the guesswork.

Reply:
@taniwha I think did mention that you should make it as flexible as possible i.e. support all combinations.